restapi/src/main/java/io/stargate/web/docsapi/service/DocumentService.java

```diff
@@ -390,6 +391,122 @@ private Object convertToBackendBooleanValue(boolean value, boolean numericBoolea
     return ImmutablePair.of(bindVariableList, firstLevelKeys);
   }
 
+  private String convertToJsonPtr(String path) {
+    return "/" + path.replaceAll(PERIOD_PATTERN.pattern(), "/").replaceAll("\\[(\\d+)\\]", "$1");
+  }
+
+  private DocumentDB maybeCreateTableAndIndexes(
+      Db dbFactory,
+      DocumentDB db,
+      String keyspace,
+      String collection,
+      Map<String, String> headers,
+      String authToken)
+      throws UnauthorizedException {
+    boolean created = db.maybeCreateTable(keyspace, collection);
+    // After creating the table, it can take up to 2 seconds for permissions cache to be updated,
+    // but we can force the permissions refetch by logging in again.
+    if (created) {
+      db = dbFactory.getDocDataStoreForToken(authToken, headers);
+      db.maybeCreateTableIndexes(keyspace, collection);
+    }
+    return db;
+  }
+
+  public List<String> writeManyDocs(
+      String authToken,
+      String keyspace,
+      String collection,
+      InputStream payload,
+      Optional<String> idPath,
+      Db dbFactory,
+      Map<String, String> headers)
+      throws IOException, UnauthorizedException {
+
+    DocumentDB db = dbFactory.getDocDataStoreForToken(authToken, headers);
+    JsonSurfer surfer = JsonSurferGson.INSTANCE;
+
+    db = maybeCreateTableAndIndexes(dbFactory, db, keyspace, collection, headers, authToken);
+    List<String> idsWritten = new ArrayList<>();
+    try (BufferedReader reader = new BufferedReader(new InputStreamReader(payload, "UTF-8"))) {
+      Iterator<String> iter = reader.lines().iterator();
+      ExecutionContext context = ExecutionContext.NOOP_CONTEXT;
+      String docsPath = convertToJsonPtr(idPath.get());
+      int chunkIndex = 0;
+      final int CHUNK_SIZE = 250;
+      while (iter.hasNext()) {
+        chunkIndex++;
+        Map<String, String> docsInChunk = new LinkedHashMap<>();
+        while (docsInChunk.size() < CHUNK_SIZE && iter.hasNext()) {
```

Comment:
Do I read it correctly that `CHUNK_SIZE` corresponds to the number of docs in each CQL batch? Why would be break input into multiple batches?

Reply:
I really don't have a reason for this; my only question being: if there are some large number of documents that someone sends, is it fine to put all of those documents in a single batch? is there a point where that would break?

If there isn't really a feasible breaking point, then I can just remove this

Reply:
Storage may have limits on max batch size, but I do not think we have visibility into that at SG level.

Since we do not really limit HTTP payload sizes for single doc inserts (do we?) it does not make batch operations worse in terms of overloading the server. It may be worth adding guardrails at HTTP level, but that's a separate concern for sure.

I believe it would not more intuitive for the end user if we had one batch per request.

If it fails at storage level, the user will have to consider using smaller batches, but at least it will be consistent with other APIs for that matter. WDYT?

Reply:
I think that's reasonable, yea - especially the point you made about how stargate doesn't have visibility into any batch size limits in storage.  Also, it's true that the single document write doesn't have a strict limit. So I will remove this!

Reply:
removed!